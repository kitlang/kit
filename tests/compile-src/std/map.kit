function main(){
    var myMap : Map[CString, Int] = Map.new(10);
    printf("Capacity: %d\n", myMap.internalArrayCapacity);
    printf("Size: %d\n", myMap.count());

    printf("First value added: %d\n", myMap.put("test1", 1));
    printf("Size: %d\n", myMap.count());

    myMap.remove("test1");
    printf("Size after removing single element: %d\n", myMap.count());

    printf("Value added: %d\n", myMap.put("test2", 3));
    printf("Value added: %d\n", myMap.put("test3", 4));
    printf("Value added: %d\n", myMap.put("test4", 5));

    printf("Key Exists 'test2': %s\n", if(myMap.exists("test2")) then "true" else "false");
    printf("Key Exists 'test3': %s\n", if(myMap.exists("test3")) then "true" else "false");
    printf("Key Exists 'test5': %s\n", if(myMap.exists("test5")) then "true" else "false");

    printf("Contains 3: %s\n", if(myMap.contains(3)) then "true" else "false");
    printf("Contains 4: %s\n", if(myMap.contains(4)) then "true" else "false");
    printf("Contains 5: %s\n", if(myMap.contains(5)) then "true" else "false");
    printf("Contains 6: %s\n", if(myMap.contains(6)) then "true" else "false");

    printf("Get 3: %d\n", myMap.get("test2").unwrap());
    printf("Get 4: %d\n", myMap.get("test3").unwrap());
    printf("Get 5: %d\n", myMap.get("test4").unwrap());
    printf("Quick Get 3: %d\n", myMap["test2"].unwrap());

    var arrayKeys : Array[CString] = myMap.keyset();
    printf("Array size of map keys: %d\n", arrayKeys.length);
    printf("Array contains 'test2': %s", if(arrayKeys[0] == "test2") then "true" else "false");

    myMap.put("1", 1);
    myMap.put("2", 1);
    myMap.put("3", 1);
    myMap.put("4", 1);
    myMap.put("5", 1);
    myMap.put("6", 1);
    myMap.put("7", 1);
    myMap.put("8", 1);
    myMap.put("9", 1);

    printf("Map resizes if more than threshold is added, capacity is double: %d\n", myMap.internalArrayCapacity);

}