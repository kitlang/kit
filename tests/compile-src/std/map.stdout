Capacity: 10
Size: 0
First value added: 1
Size after removing single element: 0
Value added: 3
Value added: 4
Value added: 5
Key Exists 'test2': true
Key Exists 'test3': true
Key Exists 'test5': false
Contains 3: true
Contains 4: true
Contains 5: true
Contains 6: false
Get 3: 3
Get 4: 4
Get 5: 5
Quick Get 3: 3
Array size of map keys: 3
Array contains 'test2': true
Map resizes if more than threshold is added, capacity is double: 20